### PR TITLE
Feature/i18n

### PR DESCRIPTION
**Pull request by Crisfole**
This addresses #29 

The only real change here is that the system now stores the controller and action before each request.

If your translation file has a `phi.path.to.controller.phi_record_name` translation (see the `en.yml` added here for a better explanation), then that translation is used when a null reason is given.

This same strategy can be used to automate the current user.

### CODE REVIEW

**Comment by wkirby** [comment]
The code looks good, but do we need any documentation changes in the `README`?

File: lib/phi_attrs/railtie.rb [hunk 1]
```diff
@@ -5,9 +5,10 @@
 
 module PhiAttrs
   class Railtie < Rails::Railtie
-    initializer 'phi_attrs.initialize' do
+    initializer 'phi_attrs.initialize' do |_app|
       ActiveSupport.on_load(:active_record) do
-        ActiveRecord::Base.send :extend, PhiAttrs
+        ActiveRecord::Base.send :extend, PhiAttrs::Model
+        ActionController::Base.send :include, PhiAttrs::Controller
```
Comment (wkirby) [inline lib/phi_attrs/railtie.rb]: So this auto-extends the controller with the `PhiAttrs` stuff?

**Comment by wkirby** [comment]
@Crisfole Does this also need to be retargeted to the `breaking/0.4.0` branch?

**Comment by egreer** [comment]
> @Crisfole Does this also need to be retargeted to the `breaking/0.4.0` branch?

This is phi_attrs not Slayer

File: lib/phi_attrs/railtie.rb (hunk 1 above)
Comment (Crisfole) [reply lib/phi_attrs/railtie.rb]: Yes.

File: lib/phi_attrs/phi_record.rb [hunk 2]
```diff
@@ -225,6 +226,36 @@ def __user_id_string(access_list)
...
         access_list.map { |c| "'#{c[:user_id]}'" }.join(',')
       end
+
+      def i18n_reason
+        controller = RequestStore.store[:phi_attrs_controller]
+        action = RequestStore.store[:phi_attrs_action]
+
+        return nil if controller.blank? || action.blank?
+
+        i18n_path = ['phi'] + __path_to_controller_and_action(controller, action)
```
Comment (egreer) [inline lib/phi_attrs/phi_record.rb]: Instead of a hard coded `'phi` we could configure this like the `log_path` so that application users can customize.

Comment (Crisfole) [reply lib/phi_attrs/phi_record.rb]: Hm. Sure...I've never seen it done that way in a rails app. It shouldn't be hard.

Comment (wkirby) [reply lib/phi_attrs/phi_record.rb]: I would say we don't need to worry about this yet. I think there's also some benefit to having this in a defined place.

Comment (Crisfole) [reply lib/phi_attrs/phi_record.rb]: Let's make that an issue: #36

**Pull request merged**